website/docs/r/virtual_machine.html.markdown

```diff
@@ -544,688 +622,491 @@ resource "vsphere_virtual_machine" "vm" {
+* `custom_attributes` - (Optional) Map of custom attribute ids to attribute value strings to set for virtual machine. Please refer to the [`vsphere_custom_attributes`][docs-setting-custom-attributes] resource for more information on setting custom attributes.
 
 [docs-setting-custom-attributes]: /docs/providers/vsphere/r/custom_attribute.html#using-custom-attributes-in-a-supported-resource
 
-~> **NOTE:** Custom attributes are unsupported on direct ESXi connections 
-and require vCenter.
-
-* `storage_policy_id` - (Optional) The UUID of the storage policy to assign to VM home directory.
-
-### CPU and memory options
-
-The following options control CPU and memory settings on the virtual machine:
-
-* `num_cpus` - (Optional) The total number of virtual processor cores to assign
-  to this virtual machine. Default: `1`.
-* `num_cores_per_socket` - (Optional) The number of cores per socket in this
-  virtual machine. The number of vCPUs on the virtual machine will be
-  `num_cpus` divided by `num_cores_per_socket`. If specified, the value
-  supplied to `num_cpus` must be evenly divisible by this value. Default: `1`.
-* `cpu_hot_add_enabled` - (Optional) Allow CPUs to be added to this virtual
-  machine while it is running.
-* `cpu_hot_remove_enabled` - (Optional) Allow CPUs to be removed to this
-  virtual machine while it is running.
-* `memory` - (Optional) The size of the virtual machine's memory, in MB.
-  Default: `1024` (1 GB).
-* `memory_hot_add_enabled` - (Optional) Allow memory to be added to this
-  virtual machine while it is running.
-
-~> **NOTE:** Certain CPU and memory hot-plug options are not available on every
-operating system. Check the [VMware Guest OS Compatibility
-Guide][vmware-docs-compat-guide] first to see what settings your guest
-operating system is eligible for. In addition, at least one `terraform apply`
-must be executed before being able to take advantage of CPU and memory hot-plug
-settings, so if you want the support, enable it as soon as possible.
+~> **NOTE:** Custom attributes requires vCenter Server and is not supported on direct ESXi host connections.
+
+* `storage_policy_id` - (Optional) The ID of the storage policy to assign to the home directory of a virtual machine.
+
+### CPU and Memory Options
+
+The following options control CPU and memory settings on a virtual machine:
+
+* `num_cpus` - (Optional) The total number of virtual processor cores to assign to the virtual machine. Default: `1`.
+
+* `num_cores_per_socket` - (Optional) The number of cores per socket in the virtual machine. The number of vCPUs on the virtual machine will be `num_cpus` divided by `num_cores_per_socket`. If specified, the value supplied to `num_cpus` must be evenly divisible by this value. Default: `1`.
+
+  * `cpu_hot_add_enabled` - (Optional) Allow CPUs to be added to the virtual machine while it is powered on.
+
+  * `cpu_hot_remove_enabled` - (Optional) Allow CPUs to be removed to the virtual machine while it is powered on.
```

Comment:
```suggestion
* `cpu_hot_remove_enabled` - (Optional) Allow CPUs to be removed to the virtual machine while it is powered on.
```

Reply:
Resolved in https://github.com/hashicorp/terraform-provider-vsphere/pull/1513/commits/3554978c2c94fa84d002e59267a6bee2732cbfed.